courses/signals.py

```diff
@@ -33,3 +39,20 @@ def handle_create_course_run_certificate(
             transaction.on_commit(
                 lambda: generate_multiple_programs_certificate(user, programs)
             )
+
+
+@receiver(
+    post_save,
+    sender=CourseRun,
+    dispatch_uid="courserun_post_save",
+)
+def handle_update_course_run(
+    sender, instance, created, **kwargs
+):  # pylint: disable=unused-argument
+    """
+    When a CourseRun model is updated, sync course run grades and generate missing certificates
+    """
+    if not created:
+        transaction.on_commit(
+            lambda: generate_course_run_certificates_for_course(instance)
```

Comment:
If a CourseRun has `self_paced = True`, we will generate certificates for eligible learners whenever the CourseRun is saved, irregardless of whether `certificate_available_date` is defined.  Is that expected?

Reply:
I don't see anything wrong with having self paced courses also available for this signal.

Reply:
Okay, that is different from how course certificates are created for self paced courses today.  I just wanted to make sure that won't surprise anyone.

Reply:
What we could do is make it very specific and say only if `certificate_availble_date` was updated then sync grades. I don't imagine course runs being updated often after their creation, but we could be more specific on what kind of update is eligible to sync grades. 
What do you think?

Reply:
So the current nightly task does what is described here: https://docs.google.com/document/d/1kw-ChTQK2YZZr8QjexhQX7P-IH5wwQeXNr4xxG0i2r8/edit#heading=h.j7mv17u378xp

You could remove the `certificate_availble_date` and `self_paced` logic from within generate_course_run_certificates_for_course() and add that into your signal.  You would then need to add a check for `self_paced = True` in `get_certificate_grade_eligible_runs()`.  But then you're not trying to make the the if statement at https://github.com/mitodl/mitxonline/blob/18c5a9fd13ec67f7e45666996647c4eaa7b6cd99/courses/api.py#L814 work for two different requirements.